### ISSUE TITLE
Add modified files to the plugin repository

### ISSUE THREAD

**Issue by gounthar** [jenkins-infra/plugin-modernizer-tool#47]
### What feature do you want to see added?

I know we still have a long way to go before creating a pull request, but I wanted to share a thought with you.

I've used the tool on one of the plugins I maintain. Here's the command I entered:

`java -jar plugin-modernizer-cli/target/jenkins-plugin-modernizer-999999-SNAPSHOT.jar --plugins android-emulator-plugin --recipes AddPluginsBom,AddCodeOwner`

The output was:

```
Starting Plugin Modernizer
Plugins: [android-emulator-plugin]
Recipes: [AddPluginsBom, AddCodeOwner]
Invoking clean phase for plugin: android-emulator-plugin
Invoking rewrite plugin for plugin: android-emulator-plugin
```

My plugin didn't have a `CODEOWNERS` file, so the recipe created one for me (which is great!).

Looking ahead, when we integrate `jgit` or any other Git tool, should we consider, as a first step, adding new files to the Git repository? This could be an excellent initial step to familiarize ourselves with whichever Git tool we choose.

What are your thoughts on this approach?

**Comment by jonesbusy** [comment]
I agree, let's keep it simple for the moment. The advantage of the `AddCodeOwner` is that it doesn't have any dependency. Plugin can be refreshed or not, it will always have the same behavior.

**Comment by jonesbusy** [comment]
First step on https://github.com/jenkinsci/plugin-modernizer-tool/pull/230

What we miss (not sure about the implementation yet)

- Comparting the 2 PluginMetaData` on the print summary at the end of the plugin (We can track: Added files (tracked ArchetypeCommonFile). Removed ArchetypeCommonFile. We can also track version changes on parent, bom and JDK versions
- Using the "git status" or "git show" command via JGit to get the summary of changes
- Having a way to mark content of `ArchetypeCommonFile as modified (probably the previous point is more generic and we can detect such common file)

**Comment by jonesbusy** [comment]
`JsonUtils` will be enhanced in https://github.com/jenkins-infra/plugin-modernizer-tool/pull/496

- Merge 2 JSON document
- Compare 2 JSON document (semantic) and return the diff as a new document

The later could be also reused to compute the PR body: https://github.com/jenkins-infra/plugin-modernizer-tool/pull/425 (right now only the title is set as a user friendly title for some recipes)
